### ISSUE TITLE
s3 error when connecting to s3 bucket.

### ISSUE THREAD

**Issue by mustafakapasi-sky** [s3tools/s3cmd#817]
Invoked as: /usr/local/bin/s3cmd -d --config=/root/.s3cfg ls s3://*****
Problem: timeout: timed out
S3cmd:   1.6.1
python:   2.7.3 (default, Feb 27 2014, 19:58:35)
[GCC 4.6.3]
environment LANG=en_US.UTF-8

Traceback (most recent call last):
  File "/usr/local/bin/s3cmd", line 2919, in <module>
    rc = main()
  File "/usr/local/bin/s3cmd", line 2841, in main
    rc = cmd_func(args)
  File "/usr/local/bin/s3cmd", line 120, in cmd_ls
    subcmd_bucket_list(s3, uri)
  File "/usr/local/bin/s3cmd", line 153, in subcmd_bucket_list
    response = s3.bucket_list(bucket, prefix = prefix)
  File "/usr/local/lib/python2.7/dist-packages/S3/S3.py", line 297, in bucket_list
    for dirs, objects in self.bucket_list_streaming(bucket, prefix, recursive, uri_params):
  File "/usr/local/lib/python2.7/dist-packages/S3/S3.py", line 324, in bucket_list_streaming
    response = self.bucket_list_noparse(bucket, prefix, recursive, uri_params)
  File "/usr/local/lib/python2.7/dist-packages/S3/S3.py", line 343, in bucket_list_noparse
    response = self.send_request(request)
  File "/usr/local/lib/python2.7/dist-packages/S3/S3.py", line 1081, in send_request
    conn = ConnMan.get(self.get_hostname(resource['bucket']))
  File "/usr/local/lib/python2.7/dist-packages/S3/ConnMan.py", line 192, in get
    conn.c.connect()
  File "/usr/lib/python2.7/httplib.py", line 757, in connect
    self.timeout, self.source_address)
  File "/usr/lib/python2.7/socket.py", line 571, in create_connection
    raise err
timeout: timed out

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions
  If the error persists, please report the
  above lines (removing any private
  info as necessary) to:
   s3tools-bugs@lists.sourceforge.net
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by canerkoroglu** [comment]
Did you try to reconfigure with ?
# s3cmd --configure

**Comment by mustafakapasi-sky** [comment]
Brilliant, that actually resolved the issue. The instances were configured to use a proxy server. Going through the logs we realised that when using s3cmd, the connection was not made using the proxy server, but were going direct.
we ran s3cmd --configure, no settings were added/changed/removed (the proxy settings were there earlier and still persisted) ...but that somehow made it work.
Thank for your help
